pkg/pillar/cmd/downloader/context.go

```diff
@@ -246,6 +247,23 @@ func (ctx *downloaderContext) registerHandlers(ps *pubsub.PubSub) error {
 	ctx.subCertObjConfig = subCertObjConfig
 	subCertObjConfig.Activate()
 
+	subContentTreeResolveConfig, err := ps.NewSubscription(pubsub.SubscriptionOptions{
+		CreateHandler: handleContentTreeResolveModify,
+		ModifyHandler: handleContentTreeResolveModify,
+		DeleteHandler: handleContentTreeResolveDelete,
+		WarningTime:   warningTime,
+		ErrorTime:     errorTime,
+		AgentName:     "volumemgr",
+		AgentScope:    types.AppImgObj,
+		TopicImpl:     types.ResolveConfig{},
+		Ctx:           ctx,
+	})
+	if err != nil {
+		return err
+	}
+	ctx.subContentTreeResolveConfig = subContentTreeResolveConfig
+	subContentTreeResolveConfig.Activate()
+
 	subAppImgResolveConfig, err := ps.NewSubscription(pubsub.SubscriptionOptions{
```

Comment:
You removed/replaced the publication of AppImgResolveStatus, but you are keeping this subscription. Why not the same?

Reply:
It is because zedmanager and volumemgr are requesting for the resolution of the tag to the resolver in the downloader.